test/sparsedir/sparse.jl

```diff
@@ -1187,9 +1187,6 @@ Ari = ceil(Int64,100*Ar)
 @test_throws ArgumentError Base.SparseArrays.normestinv(Ac,21)
 @test_throws DimensionMismatch Base.SparseArrays.normestinv(sprand(3,5,.9))
 
-@test_throws ErrorException transpose(sub(sprandn(10, 10, 0.3), 1:4, 1:4))
-@test_throws ErrorException ctranspose(sub(sprandn(10, 10, 0.3), 1:4, 1:4))
-
```

Comment:
While it's always scary to delete tests, I wasn't sure why this was something that needed testing. On this branch,
```jl
julia> A = sub(sprandn(10, 10, 0.3), 1:4, 1:4)
4x4 SubArray{Float64,2,SparseMatrixCSC{Float64,Int64},Tuple{UnitRange{Int64},UnitRange{Int64}},false}:
 0.0  0.0      -1.41166    0.0    
 0.0  0.0       0.0        0.0    
 0.0  0.0       0.628624   1.40626
 0.0  1.08286   0.0       -0.97074

julia> transpose(A)
4x4 sparse matrix with 5 Float64 entries:
        [3, 1]  =  -1.41166
        [3, 3]  =  0.628624
        [4, 3]  =  1.40626
        [2, 4]  =  1.08286
        [4, 4]  =  -0.97074

julia> full(transpose(A))
4x4 Array{Float64,2}:
  0.0      0.0  0.0        0.0    
  0.0      0.0  0.0        1.08286
 -1.41166  0.0  0.628624   0.0    
  0.0      0.0  1.40626   -0.97074
```
which seems like a rather satisfactory result. (It's even sparse!)

CC @kshyatt